src/postamble.js

```diff
@@ -436,9 +454,16 @@ if (Module['noInitialRun']) shouldRunNow = false;
 
 #if EXIT_RUNTIME == 0
 #if USE_PTHREADS
-if (!ENVIRONMENT_IS_PTHREAD) // EXIT_RUNTIME=0 only applies to default behavior of the main browser thread
+// EXIT_RUNTIME=0 only applies to default behavior of the main browser thread.
+// Otherwise EXIT_RUNTIME=0 (the default) would mean that pthreads would never
+// exit by defualt wich would break a lot of existing code.
```

Comment:
```suggestion
// exit by default which would break a lot of existing code
// (because when a pthread exits it notifies the main thread
// so it can reuse the worker).
```
I'm not sure that's right, but if not, then what is the reason?

Reply:
Sorry I meant to write "pthreads would never exit".  

If this noExitRuntime is set to true in pthreads they will not exit when main is done they would stick around and would appear to never complete (joining them would block forever).